clang/lib/3C/CastPlacement.cpp

```diff
@@ -18,107 +18,244 @@
 using namespace clang;
 
 bool CastPlacementVisitor::VisitCallExpr(CallExpr *CE) {
-  auto *FD = dyn_cast_or_null<FunctionDecl>(CE->getCalleeDecl());
-  if (FD && Rewriter::isRewritable(CE->getExprLoc())) {
-    // Get the constraint variable for the function.
-    FVConstraint *FV = Info.getFuncConstraint(FD, Context);
-    // Function has no definition i.e., external function.
+  // Get the constraint variable for the function.
+  Decl *CalleeDecl = CE->getCalleeDecl();
+  FVConstraint *FV = nullptr;
+  FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(CalleeDecl);
+  if (FD) {
+    FV = Info.getFuncConstraint(FD, Context);
     assert("Function has no definition" && FV != nullptr);
+  } else if (isa_and_nonnull<VarDecl>(CalleeDecl)) {
+    CVarOption Opt = Info.getVariable(cast<VarDecl>(CalleeDecl), Context);
+    if (Opt.hasValue()) {
+      if (isa<PVConstraint>(Opt.getValue()))
+        FV = cast<PVConstraint>(Opt.getValue()).getFV();
+      else
+        FV = dyn_cast<FVConstraint>(&Opt.getValue());
+    }
+  }
```

Comment:
So I could end up here with `FV == nullptr` because I have something like `foo[0](a,b)` -- what should happen in this case?

Reply:
Put another way: Should we be asserting something here? I suppose it's fine to do nothing if `FV` is `nullptr` but that would seem odd.